src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -852,7 +873,13 @@ - (nullable MTRDeviceController *)runningControllerForFabricIndex:(chip::FabricI
 
 - (void)operationalInstanceAdded:(chip::PeerId &)operationalID
 {
-    for (MTRDeviceController * controller in _controllers) {
+    assertChipStackLockedByCurrentThread();
```

Comment:
I'm wondering if we should add an "assertNotRunningOnMatterQueue" and append it to all methods that are expected to run on the user thread (basically most of those).
Many of the methods of the controller factory will deadlock if called from the Matter queue, but some (e.g `runningControllerForFabricIndex` which is called from the Matter queue by `MTROTAProviderDelegateBridge::OnTransferSessionBegin`) won't and me expose the same issue no ?